src/main/java/seedu/guilttrip/logic/parser/remindercommandparsers/AddGeneralReminderCommandParser.java

```diff
@@ -0,0 +1,76 @@
+package seedu.guilttrip.logic.parser.remindercommandparsers;
+
+import static seedu.guilttrip.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_DESC;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_END_DATE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_LOWER_BOUND;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_START_DATE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_TYPE;
+import static seedu.guilttrip.logic.parser.CliSyntax.PREFIX_UPPER_BOUND;
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import seedu.guilttrip.logic.commands.remindercommands.AddGeneralReminderCommand;
+import seedu.guilttrip.logic.parser.ArgumentMultimap;
+import seedu.guilttrip.logic.parser.ArgumentTokenizer;
+import seedu.guilttrip.logic.parser.Parser;
+import seedu.guilttrip.logic.parser.ParserUtil;
+import seedu.guilttrip.logic.parser.Prefix;
+import seedu.guilttrip.logic.parser.exceptions.ParseException;
+import seedu.guilttrip.model.entry.Description;
+import seedu.guilttrip.model.tag.Tag;
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class AddGeneralReminderCommandParser implements Parser<AddGeneralReminderCommand> {
+    public static final String INCORRECT_AMT_RANGE = "Quota lower bound must be below the upper bound";
+    public static final String INCORRECT_DATE_RANGE = "Starting date must be before ending date";
+
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddGeneralReminderCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(
+                        args, PREFIX_TYPE, PREFIX_DESC, PREFIX_UPPER_BOUND, PREFIX_LOWER_BOUND,
+                        PREFIX_START_DATE, PREFIX_END_DATE, PREFIX_TAG);
+        if (!arePrefixesPresent(argMultimap, PREFIX_DESC)) {
```

Comment:
[Optional]: You can use the `throwError....()` function in `ParserUtil.java` for more expressive error messages. It's up to you here